### PR TITLE
Data Export - Controller

### PR DESCRIPTION
**Pull request by ChrisHuynh333**
## What does this PR do and why?
This PR adds the `data_export_controller`.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
1. Ensure tests pass locally
Note: Many expected tests require a template view. These tests will be added as the data export UI gets added

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/controllers/data_exports_controller.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+class DataExportsController < ApplicationController
+  before_action :data_export, only: %i[download destroy]
+
+  def index
+    @data_exports = DataExport.where(user: current_user)
+  end
+
+  def new
+    render turbo_stream: turbo_stream.update('export_modal',
+                                             partial: 'new_export_modal',
```
Comment (joshsadam) [inline app/controllers/data_exports_controller.rb:13]: **question:** Is the UI (and this partial) coming in a subsequent PR?

Comment (ChrisHuynh333) [reply app/controllers/data_exports_controller.rb:13]: Yes all UI elements will be tied in with future PRs

Comment (joshsadam) [reply app/controllers/data_exports_controller.rb:13]: Ok that's what I figured.  My only comment here is that you should not be referring to UI elements that do not yet exist. Don't worry about it now as I suspect you have another branch with this already.

**Review by joshsadam — APPROVED** [review]
This looks good to me.

**Pull request merged**